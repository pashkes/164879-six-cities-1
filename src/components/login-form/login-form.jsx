import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import DocumentTitle from "react-document-title";

import {Operation} from "../../reducer/user/user";
import {withLoginForm} from "../../hocs/with-login-form/with-login-form";
import {getError} from "../../reducer/user/selectors";

export class LoginForm extends PureComponent {
  constructor(props) {
    super(props);

    this._handleSubmitForm = this._handleSubmitForm.bind(this);
  }

  _handleSubmitForm(evt) {
    evt.preventDefault();
    const {
      emailValue,
      passwordValue,
      onLogIn,
    } = this.props;

    onLogIn(emailValue, passwordValue);
  }

  render() {
    const {
      onChangeEmail,
      onChangePassword,
      emailValue,
      passwordValue,
      error,
    } = this.props;

    return (
      <DocumentTitle title="Page of log in">
        <section className="login">
          <h1 className="login__title">Sign in</h1>
          <form onSubmit={this._handleSubmitForm} className="login__form form" action="#" method="post">
            <div className="login__input-wrapper form__input-wrapper">
              <label htmlFor="login-email" className="visually-hidden">E-mail</label>
              <input
                onChange={onChangeEmail}
                value={emailValue}
                id="login-email"
                className="login__input form__input"
                type="email"
                name="email"
                placeholder="Email"
                pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$"
                required
              />
            </div>
            <div className="login__input-wrapper form__input-wrapper">
              <label htmlFor="login-password" className="visually-hidden">Password</label>
              <input
                onChange={onChangePassword}
                value={passwordValue}
                id="login-password"
                className="login__input form__input"
                type="password"
                name="password"
                placeholder="Password"
                required
              />
            </div>
            <span role="alert" aria-live="polite">{error}</span>
            <button className="login__submit form__submit button" type="submit">Sign in</button>
          </form>
        </section>
      </DocumentTitle>
    );
  }
}

LoginForm.propTypes = {
  onChangeEmail: PropTypes.func.isRequired,
  onChangePassword: PropTypes.func.isRequired,
  emailValue: PropTypes.string,
  passwordValue: PropTypes.string,
  onLogIn: PropTypes.func.isRequired,
  error: PropTypes.string,
};

const mapStateToProps = (state) => ({
  error: getError(state),
});
const mapDispatchToProps = (dispatch) => ({
  onLogIn: (email, password) => dispatch(Operation.authorization(email, password)),
});

export default connect(mapStateToProps, mapDispatchToProps)(withLoginForm(LoginForm));
