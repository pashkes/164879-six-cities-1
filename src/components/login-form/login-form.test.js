import React from 'react';
import {LoginForm} from "./login-form.jsx";
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

it(`Login form renders correctly`, () => {
  const cards = shallow(<LoginForm
    onChangeEmail={jest.fn()}
    onChangePassword={jest.fn()}
    logIn={jest.fn()}
  />);
  expect(cards).toMatchSnapshot();
});
