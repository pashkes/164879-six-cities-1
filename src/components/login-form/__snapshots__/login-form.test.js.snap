// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form renders correctly 1`] = `
<SideEffect(DocumentTitle)
  title="Page of log in"
>
  <section
    className="login"
  >
    <h1
      className="login__title"
    >
      Sign in
    </h1>
    <form
      action="#"
      className="login__form form"
      method="post"
      onSubmit={[Function]}
    >
      <div
        className="login__input-wrapper form__input-wrapper"
      >
        <label
          className="visually-hidden"
          htmlFor="login-email"
        >
          E-mail
        </label>
        <input
          className="login__input form__input"
          id="login-email"
          name="email"
          onChange={[MockFunction]}
          pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,4}$"
          placeholder="Email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="login__input-wrapper form__input-wrapper"
      >
        <label
          className="visually-hidden"
          htmlFor="login-password"
        >
          Password
        </label>
        <input
          className="login__input form__input"
          id="login-password"
          name="password"
          onChange={[MockFunction]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <span
        aria-live="polite"
        role="alert"
      />
      <button
        className="login__submit form__submit button"
        type="submit"
      >
        Sign in
      </button>
    </form>
  </section>
</SideEffect(DocumentTitle)>
`;
